torch/_inductor/scheduler.py

```diff
@@ -3865,7 +3865,7 @@ def score_fusion_memory(
         memory operations.
         """
         node1_dep_len = len(node1.read_writes.reads) + len(node1.read_writes.writes)
-        node2_dep_len = len(node1.read_writes.reads) + len(node2.read_writes.writes)
+        node2_dep_len = len(node2.read_writes.reads) + len(node2.read_writes.writes)
 
         # optimization: iter over smaller set
         if min(node1_dep_len, node2_dep_len) * 4 < max(node1_dep_len, node2_dep_len):
```

Comment:
Would you mind just removing this optimization? i think we had a similar pr somewhere. This shouldnt actually affect the scheduler fusions since in either case we will do compute the full result.